goto/tests/test_end_to_end.sh

```diff
@@ -217,26 +217,57 @@ function test_06_goto_add {
     _projectfile_should_contain "test_escaped_ampersand_in_url"
 }
 
-function test_07_goto {
-    magicword="test_goto"
-    testcdpath="/tmp"
-    nonexisting_magicword="IDoNotExist"
-    uri="http://example.com"
+function test_07_goto_many {
+    declare -a magicwords
+    declare -a uris
+    magicwords=(
+        "red"
+        "blue"
+        "green"
+        "yellow"
+        "orange"
+    )
+    uris=(
+        "\"http://red.com\""
```

Comment:
I do indeed see the need to test multiple schemes. 
However I think we should think a little more about using these schemes in the automated tests for now ..
They should work in Circle Ci, which runs them in a linux docker container (and a linux VM) without any desktop.

They will causesideeffects from running the tests - opening windows of different kinds.
I'm not sure they would all work in different environments. 

And this also binds goto developers to have spotify installed -- I kinda don't want to tie goto to any particular other software

Reply:
Alright. I didn't know the tests actually opens windows.

Reply:
Don't misunderstand, these are powerful tests. 
It might be a possibility to run this test conditionally - if explicitly setting an environment wariable or a command line argument. 

The side-effect of opening multiple windows should be optional  on devs machines -- and I hope we can find a viable solution for making  this run and verify the results in  Circle CI. 

You kinda hit a nerve here -- or an effect of goto that is really important to test - but hard to test within the current framework without causing side-effects and making sure the tests will run on different environments.

Reply:
Actually, they should open windows, if they worked ^^

Reply:
`make test` has never opened a window for me

Reply:
> You kinda hit a nerve here -- or an effect of goto that is really important to test - but hard to test within the current framework without causing side-effects and making sure the tests will run on different environments.

I thought for a second about "maybe I should not do this?", but then some other person in my head said "You should do whatever you want. It's a test! It is meant to break things. Make a mess!"

Reply:
hehe - I think that if no windows are opened - it is actually a symptom of a misbehaving goto. 
Then goto should `exit 1` for sure.

Reply:
I think we consider fixing this as outside the scope of this PR - as it is really two issues:

 1. an issue with goto on failing open/start calls. 
2. an issue with the test system that should actually cause side effects if calling these commands.